backend/core/src/main/java/org/sonarsource/sonarlint/core/plugin/PluginsService.java

```diff
@@ -120,6 +155,11 @@ private Set<Path> getPluginPathsForConnection(String connectionId) {
     // order is important as e.g. embedded takes precedence over stored
     pluginsToLoadByKey.putAll(pluginsStorage.getStoredPluginPathsByKey());
     pluginsToLoadByKey.putAll(getEmbeddedPluginPathsByKey(connectionId));
+    if (shouldUseEnterpriseCSharpAnalyzer(connectionId) && csharpSupport.csharpEnterprisePluginPath != null) {
```

Comment:
Nitpick: Coverage can be improved here, but it would require mocking a lot of things.